src/Discord.py

```diff
@@ -0,0 +1,69 @@
+import requests, json
+from src.Dev import Dev
+from src.Settings import Settings
+from src.TERMGUI.Log import Log
+
+from discord_webhook import DiscordWebhook
+
+
+class Discord:
+
+    dev_endpoint = Settings.get_key(Settings.discord_dev_key)
+    prod_endpoint = Settings.get_key(Settings.discord_prod_key)
+
+    def __init__(self, content = "", quiet = False):
+        self.content = content
+        self.quiet = quiet
+
+        self.dev = True if Dev.isDev() else False
+        self.username = Settings.get_username(capitalize = True)
+
+        self.set_endpoint()
+
+    def set_endpoint(self):
+        if self.dev:
+            endpoint = self.dev_endpoint
+        else:
+            endpoint = self.prod_endpoint
+
+        self.endpoint = endpoint
+
+    def post(self):
+        webhook = self.build_webhook()
+
+        if not self.quiet:
+            Log("Sending Discord notification")
+
+        response = webhook.execute()
+
+        if not self.quiet and not response.ok:
+            Log("Notification webhook failed:")
+            Log(f"  Webhook: {inspect.getmembers(webhook)}")
+            Log(f"  Response: {inspect.getmembers(response)}")
+
+    def build_webhook(self):
+        return DiscordWebhook(
+                url = self.endpoint,
+                rate_limit_retry = True,
+                content = self.content)
+
+    def post_link(self, link_title, url):
+        description = f"{self.username} uploaded {link_title}"
+
+        self.content = f"{description}: {url}"
+
+        self.post()
+
+    def post_message(self, content):
+        self.content = content
+
+        self.post()
+
+    def post_log(self, content, quiet = True):
+        log = f"{self.username} – Log: {content}"
```

Comment:
Not a big one but if we can separate the "<username> - Log:" line and start the content on the next line?  So that the log file's first line isn't indented with the username.

Reply:
Would this suffice?

```suggestion
        log = f"Log – {self.username}:\n\n{content}"
```

My thought was that it would be good to have the username and log content all in the message, so that it's never unclear whose log you're reading in Discord, and so that we can avoid making two HTTP requests. But this is definitely a micro optimization, and I have no strong feelings either way. Let me know if you prefer it how you had it originally with the two requests back-to-back.

Reply:
So I went ahead and made [this change](https://github.com/vindennl48/LOFSongManager/pull/10/commits/76f191c1ff90b83e61ad178090642f09603546ab) just to get it in there tonight, but still let me know if you'd prefer something different.